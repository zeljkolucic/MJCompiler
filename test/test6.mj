program MJProgram
	int x; int y; const int one = 1; int i; bool flag;
{

	void main() 
	{
		x = 5;
		i = 0;
		flag = true;
		
		print(x); 
		
		do {
			x++;
			i++;
			if(x % 8 == 0)
				break;		
		} while(i < 10 && flag);
		
		print(x);
		
		y = 16;
		
		if(x >= y + one) {
			print(1);
		} else {
			print(-1);
		}
		
		do {
			x++;
			
			if(x >= 10)
				continue;
				
			y--;
			
		} while(x <= y + one);
		
		print(x);
		print(y);
		
	}

}