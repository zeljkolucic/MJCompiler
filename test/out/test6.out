Linija [7]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [8]: Pristup globalnoj promenljivoj i.
DesignatorIdent(
 i
) [DesignatorIdent]
Linija [9]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [11]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [14]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [15]: Pristup globalnoj promenljivoj i.
DesignatorIdent(
 i
) [DesignatorIdent]
Linija [16]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [18]: Pristup globalnoj promenljivoj i.
DesignatorIdent(
 i
) [DesignatorIdent]
Linija [18]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [20]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [22]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [24]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [24]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [31]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [33]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [36]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [38]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [38]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [40]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [41]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Var x: int, 0, 0 
   Var y: int, 1, 0 
   Con one: int, 1, 0 
   Var i: int, 2, 0 
   Var flag: , 3, 0 
   Meth main: notype, 0, 0 



